### PR TITLE
KPMP-4032-Address-library-vulnerabilities

### PR DESCRIPTION
**Pull request by abarkellumich**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,11 +34,10 @@
     "react-ga4": "1.4.0",
     "react-redux": "7.2.4",
     "react-router-dom": "5.2.1",
-    "react-scripts": "4.0.3",
     "reactstrap": "8.9.0",
     "redux": "4.1.1",
     "redux-thunk": "2.3.0",
-    "vitessce": "1.1.21"
+    "sass": "1.56.1"
```
Comment (abarkellumich) [inline package.json:40]: node-sass-chokidar is now deprecated, and the recommendation is to switch over to dart-sass.

File: src/App.js [hunk 2]
```diff
@@ -12,8 +12,7 @@ import ErrorBoundaryContainer from './components/Error/ErrorBoundaryContainer';
 import Oops from './components/Error/Oops';
 import NotFoundPage from './components/Error/NotFoundPage';
 import ImageDatasetListContainer from "./components/SpatialViewer/ImageDatasetListContainer";
-import SpatialViewerContainer from "./components/SpatialViewer/SpatialViewerContainer";
-import { baseURL } from '../package.json';
+import packagejson from '../package.json';
```
Comment (abarkellumich) [inline src/App.js:15]: With latest version of react-scripts, we can no longer deconstruct objects on import

File: src/App.js [hunk 3]
```diff
@@ -52,8 +51,8 @@ store.subscribe(saveState);
 
 const connector = new AppSearchAPIConnector({
   searchKey: process.env.REACT_APP_SEARCH_KEY,
-  engineName: "spatial-viewer",
   endpointBase: "/spatial-viewer/search",
+  engineName: "atlas-repository",
```
Comment (abarkellumich) [inline src/App.js:55]: Not strictly related to vulns, but need to point at correct engine

File: src/components/SpatialViewer/SpatialViewer.js [hunk 4]
```diff
@@ -1,101 +0,0 @@
-import React, { Component } from 'react';
```
Comment (abarkellumich) [inline src/components/SpatialViewer/SpatialViewer.js:1]: Removing spatial viewer as it isn't needed for the repository, and the vitessce package has a number of vulnerabilities associated with it.

File: src/components/SpatialViewer/SpatialViewerContainer.js [hunk 5]
```diff
@@ -1,39 +0,0 @@
-import {connect} from "react-redux";
-import SpatialViewer from "./SpatialViewer";
```
Comment (abarkellumich) [inline src/components/SpatialViewer/SpatialViewerContainer.js:2]: Removing spatial viewer as it isn't needed for the repository, and the vitessce package has a number of vulnerabilities associated with it.

File: src/App.js [hunk 6]
```diff
@@ -87,12 +86,11 @@ class App extends Component {
     return (
       <Provider store={store}>
         <SearchProvider config={searchConfig}>
-        <BrowserRouter history={history} basename={baseURL}>
+        <BrowserRouter history={history} basename={packagejson.baseURL}>
           <ErrorBoundaryContainer>
             <NavBar app='atlas' />
             <Switch>
               <Route exact path="/" component={ImageDatasetListContainer} store={store} />
-              <Route exact path="/view" component={SpatialViewerContainer} store={store} />
```
Comment (abarkellumich) [inline src/App.js:95]: Removing spatial viewer as it isn't needed for the repository, and the vitessce package has a number of vulnerabilities associated with it.

**Review by rlreamy — APPROVED** [review]

**Pull request merged**